tests/unittests/GraphTest.cpp

```diff
@@ -1266,11 +1266,95 @@ TEST(Graph, verifyConstantTensorTypeMatchesTensorTypeChanged) {
   auto *input = M.createConstant(ElemKind::FloatTy, {5}, "input");
   // Fresh constant should verify just fine.
   EXPECT_TRUE(input->verify());
-  input->getPayload().convertToType(ElemKind::Float16Ty);
+  input->getPayloadMutable().convertToType(ElemKind::Float16Ty);
 
   EXPECT_FALSE(input->verify());
 }
 
+/// Check that Constants backed by unowned Tensors are in fact unowned until
+/// a mutable reference to their payload is obtained at which point the backing
+/// Tensor is copied and becomes owned.
+TEST(Graph, verifyConstantWithUnownedTensorCopiesOnWrite) {
+  Module M;
+
+  Tensor originalT(ElemKind::FloatTy, {3});
+  Tensor unownedT = originalT.getUnowned({3});
+
+  auto originalH = originalT.getHandle();
+  auto unownedH = unownedT.getHandle();
+
+  for (size_t i = 0; i < originalT.size(); i++) {
+    originalH.raw(i) = i;
+  }
+
+  // Both Tensors should have the same values because unownedH has the same
+  // underlying memory as originalH.
+  for (size_t i = 0; i < unownedH.size(); i++) {
```

Comment:
nit: could just call `Tensor::isEqual()`.

Reply:
Checking that `Tensor::getUnsafePtr()` are equal would probably be a good check too, even if it's somewhat redundant. Could be done after getting the mutable payload too, to verify that the pointers are now different.